meteor-slingshot/lib/upload.js

```diff
@@ -202,13 +202,22 @@ Slingshot.Upload = function (directive, metaData) {
         );
       });
 
-      xhr.open('POST', self.instructions.upload, true);
+      let isR2 = self.instructions.upload.includes('r2.cloudflarestorage.com');
```

Comment:
Isn't it better to add a flag saying what's the service that this instructions are from? It comes from the service upload function, so you would just need to add the flag there and check here. Seems better than check by the url.

Reply:
Also, you can use `const` here

Reply:
fixed